server/finder/handler/finder_handler.go

```diff
@@ -247,16 +250,19 @@ func (h *FinderHandler) GetProvider(providerID peer.ID) ([]byte, error) {
 }
 
 func (h *FinderHandler) GetStats() ([]byte, error) {
-	size, err := h.indexer.Size()
-	if err != nil {
+	stats, err := h.stats.get()
+	switch {
+	case err != nil:
 		return nil, err
+	case stats.EntriesEstimate == 0:
+		return nil, nil
+	default:
+		return model.MarshalStats(&stats)
 	}
+}
 
-	s := model.Stats{
-		EntriesEstimate: size / avg_mh_size,
```

Comment:
should we add this as a new field - so we keep the current `EntriesEstimate` but also add a `EntriesCount` with the more precise stats from the engine?

Reply:
Both are estimates. The existing one is very off and backing store-dependant.
I am curious what value would that deliver?

Reply:
it would allow a period where we could report both metrics and say what the correction factor we're applying is in switching to the new one.

Reply:
That's fine; i'll add that as a separate commit so that we can revert it easily when we do the comparison study.

Reply:
See: https://github.com/filecoin-project/storetheindex/pull/1074/commits/9ec2de45935dc01a68a09eaea2022cbde9e5f50c